config/proposer/loader/loader.go

```diff
@@ -117,8 +129,11 @@ func (psl *settingsLoader) Load(cliCtx *cli.Context) (*proposer.Settings, error)
 	loadConfig := &validatorpb.ProposerSettingsPayload{}
 
 	// override settings based on other options
-	if psl.options.builderConfig != nil && psl.options.gasLimit != nil {
-		psl.options.builderConfig.GasLimit = *psl.options.gasLimit
+	if psl.options.builderConfig != nil {
```

Comment:
Will panic if `psl.options == nil`.

Reply:
i don't think options can be nil but i'll add a check anyways